misc/dbt-materialize/dbt/include/materialize/macros/deploy/deploy_promote.sql

```diff
@@ -136,3 +140,52 @@ COMMIT;
         {{ return(count.rows[0][0]) }}
     {% endif %}
 {% endmacro %}
+
+{% macro get_current_database() %}
+    {% set current_database_query = "SELECT current_database()" %}
+    {% set results = run_query(current_database_query) %}
+    {% if execute %}
+        {% set current_database = results.rows[0][0] %}
+        {{ return(current_database) }}
+    {% else %}
+        {{ return(None) }}
+    {% endif %}
+{% endmacro %}
+
+{% macro discover_sinks() %}
+    {% set sinks_to_alter = [] %}
+    {% for node in graph.nodes.values() | selectattr("resource_type", "equalto", "model") | selectattr("config.materialized", "equalto", "sink") %}
+        {% set sink_database = node.database %}
+        {% set sink_schema = node.schema %}
+        {% set sink_name = node.name %}
+        {% set upstream_node = graph.nodes[node.depends_on.nodes[0]] %}
+        {% set new_upstream_relation = adapter.quote(upstream_node.database) ~ '.' ~ adapter.quote(upstream_node.schema) ~ '.' ~ adapter.quote(upstream_node.name) %}
+        {% set sink = {
+            "database": sink_database,
+            "schema": sink_schema,
+            "name": sink_name,
+            "new_upstream_relation": new_upstream_relation
+        } %}
+        {% do sinks_to_alter.append(sink) %}
```

Comment:
IIUC, this will consider _all_ sinks in the graph. Wouldn't we want to only consider sinks that have an upstream dependency in the schemas and clusters affected by the blue/green deployment?

Reply:
Maybe I got the wrong impression here:

> I think it should be a no-op though to do an `ALTER SINK s SET FROM <same from>`. Might help cut down on the complexity in the dbt adapter. If it’s not a cheap no-op in Materialize, we should make it one!

But we could always add an extra check in this macro to verify if the upstream object is in the var.deployment.schemas list.

Reply:
There [can be consequences](https://materialize.com/docs/sql/alter-sink/#cutover-scenarios) to cutting over a sink, so I wouldn't just cutover all sinks in the graph _unless_ they depend on a changed object.

Reply:
Or, an object that is part of the blue/green deployment.

Reply:
Makes sense! Just added an additional check here to verify if the upstream object that the sink uses, is actually part of the deployment schemas. Let me know what you think!